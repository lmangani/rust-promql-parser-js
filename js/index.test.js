const { promql_parse } = require("../pkg/promql_parser_js.js");

describe('parse_promql', () => {
  test('convert promql to json ast', () => {

    const query = 'sum(rate(foo{bar="baz"}[5m])) by (x,y)';
    const result = promql_parse(query);
    expect(result).toEqual({
      '@type': 'aggregate',
      op: 'sum',
      modifier: {
        include: ['x', 'y'],
      },
      param: null,
      expr: {
        '@type': 'call',
        function: {
          name: 'rate',
          arg_types: ['matrix'],
          return_type: 'vector',
          variadic: false,
        },
        args: [
          {
            '@type': 'matrix_selector',
            range: 300,
            offset: null,
            vector: {
              '@type': 'vector_selector',
              name: 'foo',
              matchers: [
                { name: 'bar', op: '=', value: 'baz' },
              ],
              at: null,
            },
          },
        ],
      },
    });
  });
});
