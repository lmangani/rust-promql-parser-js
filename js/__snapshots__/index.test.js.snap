// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse_promql convert promql to json ast 1`] = `
Object {
  "@type": "aggregate",
  "expr": Object {
    "@type": "call",
    "args": Array [
      Object {
        "@type": "matrix_selector",
        "range": 300,
        "vector": Object {
          "@type": "vector_selector",
          "at": null,
          "matchers": Array [
            Object {
              "name": "bar",
              "op": "=",
              "value": "baz",
            },
          ],
          "name": "foo",
          "offset": null,
        },
      },
    ],
    "function": Object {
      "arg_types": Array [
        "matrix",
      ],
      "name": "rate",
      "return_type": "vector",
      "variadic": false,
    },
  },
  "modifier": Object {
    "include": Array [
      "x",
      "y",
    ],
  },
  "op": "sum",
  "param": null,
}
`;
